### ISSUE TITLE
Clf loss is much larger than Res Loss

### ISSUE THREAD

**Issue by keshawnhsieh** [zylo117/Yet-Another-EfficientDet-Pytorch#274]
I was training with my own dataset, which has 50,000 images totally and I split it into 85%:15% to train and val. And I have to train model from scratch for some reasons. I start running with
```
python train.py -c 3 -p icartoonface \
--batch_size 12 \
--optim sgd \
--lr 8e-2 \
--num_epochs 10 \
--data_path ../data \

```
One night past, I got such logs
```
Step: 3540. Epoch: 0/10. Iteration: 3541/3541. Cls loss: 2.30212. Reg loss: 0.08922. Total loss: 2.39134: 100%|█| 3541/3541 [1:12:08<
Val. Epoch: 0/10. Classification loss: 2.28463. Regression loss: 0.09227. Total loss: 2.37690
Step: 7081. Epoch: 1/10. Iteration: 3541/3541. Cls loss: 2.30212. Reg loss: 0.09391. Total loss: 2.39603: 100%|█| 3541/3541 [1:11:47<
Val. Epoch: 1/10. Classification loss: 2.28463. Regression loss: 0.09094. Total loss: 2.37557
Step: 10622. Epoch: 2/10. Iteration: 3541/3541. Cls loss: 2.30212. Reg loss: 0.09736. Total loss: 2.39949: 100%|█| 3541/3541 [1:12:01
Val. Epoch: 2/10. Classification loss: 2.28463. Regression loss: 0.09089. Total loss: 2.37552
Step: 14163. Epoch: 3/10. Iteration: 3541/3541. Cls loss: 2.30212. Reg loss: 0.09228. Total loss: 2.39441: 100%|█| 3541/3541 [1:12:22
Val. Epoch: 3/10. Classification loss: 2.28463. Regression loss: 0.09079. Total loss: 2.37542
Step: 17704. Epoch: 4/10. Iteration: 3541/3541. Cls loss: 2.30212. Reg loss: 0.08682. Total loss: 2.38894: 100%|█| 3541/3541 [1:12:16
Val. Epoch: 4/10. Classification loss: 2.28463. Regression loss: 0.09063. Total loss: 2.37526
Step: 21245. Epoch: 5/10. Iteration: 3541/3541. Cls loss: 2.30212. Reg loss: 0.08756. Total loss: 2.38968: 100%|█| 3541/3541 [1:12:22
Val. Epoch: 5/10. Classification loss: 2.28463. Regression loss: 0.09081. Total loss: 2.37544
Step: 24786. Epoch: 6/10. Iteration: 3541/3541. Cls loss: 2.30212. Reg loss: 0.09623. Total loss: 2.39835: 100%|█| 3541/3541 [1:12:02
Val. Epoch: 6/10. Classification loss: 2.28463. Regression loss: 0.09106. Total loss: 2.37569
```
I notice that loss seems to be stuck at some stage, and classification loss is much larger than regression loss. Is such imbalance between losses normal? And how can I figure out to make loss decrease again? I tried to reduce the learning rate and resume training with `argument --load_weights last ` but it seems have no help.

**Comment by zylo117** [comment]
Try training with `--head_only True` and much smaller lr.

And cls loss is much greater than reg loss, about 10 times.

**Comment by keshawnhsieh** [comment]
> Try training with `--head_only True` and much smaller lr. Cls loss 2.3 means classification is zero-out and not learning anything.
> 
> And cls loss is much greater than reg loss, about 10 times.

For some reasons, I can't use any Pretrained Weights. So only training with head maybe not suitable for me?

**Comment by zylo117** [comment]
Why not? The performance will be bad if you don't use pretrained weights.
Even the datasets are not similar, without any pretrained weights is a bad idea.

**Comment by Pari-singh** [comment]
I am facing an issue where the regression loss is not changing but the classification loss is decreasing over iterations. The reg loss is stuck at around 5. I am training just the head. Any comments? @zylo117

**Comment by zylo117** [comment]
> I am facing an issue where the regression loss is not changing but the classification loss is decreasing over iterations. The reg loss is stuck at around 5. I am training just the head. Any comments? @zylo117

If gt boxes are strange shapes like extremely long rectangles, then reg loss is not likely going to drop easily. You can tune the anchor config to get a better performance though.
